litgpt/scripts/merge_lora.py

```diff
@@ -43,16 +43,19 @@ def merge_lora(
     fabric = L.Fabric(devices=1, precision=precision, accelerator="cpu")
     config = Config.from_file(checkpoint_dir / "model_config.yaml", **lora_params)
 
-    with fabric.init_module():
+    with fabric.init_module(), torch.device("meta"):
         model = GPT(config)
 
     lora_path = checkpoint_dir / "lit_model.pth.lora"
     pretrained_checkpoint = torch.load(str(pretrained_checkpoint_dir / "lit_model.pth"), mmap=True)
     lora_checkpoint = torch.load(str(lora_path), mmap=True)
+    # since LoRA finetuning only saves the LoRA weights, we treat the lora weights dtype as the expected dtype
+    lora_dtype = next(iter(lora_checkpoint.values())).dtype
+    pretrained_checkpoint = {k: v.to(dtype=lora_dtype) for k, v in pretrained_checkpoint.items()}
```

Comment:
Can you point out what is actually now the difference of this vs how it is in main? Now that this conversion happens, mmap is not doing anything anymore. Same with assign: Before, `load_state_dict` would do the conversion and now we do it manually, what is the advantage? And there is actually no use for the meta device anymore now.

Reply:
I had to do the conversion later

Reply:
Sure ok. In that sense, to answer my question, the main advantage is that we don't save the weights twice in memory, since the model's weights will be on the meta device. But other than the rest is still equivalent. Can you convirm?

Reply:
I don't know what you mean by "the rest".

Yes, the entire point of assign is to not have two copies in memory, that's right.

The `.to()` call is to avoid the issue caused by us not saving the pretrained weights during fine-tuning.

Reply:
By the rest I mean the conversion step, which is done implicitly in main right now.
In a previous version, you did the conversion over the dictionary. So if this PR didn't save memory, naturally the question was what is the benefit? 
So in that sense, it doesn't revert what we've fixed in https://github.com/Lightning-AI/litgpt/commit/d8dc97e4160307a674a6f4bd3ccdacbaa90b4c90, but instead it reimplements #1189.

Reply:
Because the title of this PR is "Ensure pretrained and lora weights have the same dtype in merge_lora" which we already did in https://github.com/Lightning-AI/litgpt/commit/d8dc97e4160307a674a6f4bd3ccdacbaa90b4c90. That's what's confusing me.